File: kitten-house-care/src/main/java/org/jesperancinha/housing/controller/CatControllerImpl.java
package org.jesperancinha.housing.controller;

import org.jesperancinha.housing.data.CatDto;
import org.jesperancinha.housing.service.CatServiceImpl;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

@RestController
@RequestMapping("/cats")
public class CatControllerImpl {

    private final CatServiceImpl catService;

    public CatControllerImpl(CatServiceImpl catService) {
        this.catService = catService;
    }

    @org.springframework.web.bind.annotation.GetMapping
    public Flux<CatDto> getAllCats() {
        return this.catService.getAllCats();
    }

    @org.springframework.web.bind.annotation.GetMapping("/{catId}")
    public Mono<CatDto> getCatByIdI(@org.springframework.web.bind.annotation.PathVariable Long catId) {
        return this.catService.getCatById(catId);
    }

    @org.springframework.web.bind.annotation.GetMapping("/full/{catId}")
    public Mono<CatDto> getFullCatById(@org.springframework.web.bind.annotation.PathVariable Long catId) {
        return this.catService.getFullCatById(catId);
    }

    @org.springframework.web.bind.annotation.GetMapping("/full")
    public Flux<CatDto> getFullAllCats() {
        return this.catService.getFullAllCats();
    }

    @org.springframework.web.bind.annotation.GetMapping("/fullForTest")
    public Flux<CatDto> getFullAllCatsReactiveForTest() {
        return this.catService.getFullAllCats().map(catDto -> {
            try {
                Thread.sleep(1);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            return catDto;
        });
    }

    @org.springframework.web.bind.annotation.GetMapping("/full/nonreactive")
    public List<CatDto> getFullAllCatsNonReactive() {
        try {
            Thread.sleep(1);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return this.catService.getFullAllCatsNonReactive();
    }
}

File: kitten-house-care/src/main/kotlin/org/jesperancinha/housing/controller/CatControllerImpl.kt
package org.jesperancinha.housing.controller

import org.jesperancinha.housing.data.CatDto
import org.jesperancinha.housing.service.CatServiceImpl
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/cats")
class CatControllerImpl(private val catService: CatServiceImpl) {
    @get:GetMapping
    val allCats: Flux<CatDto?>?
        get() = catService.allCats

    @GetMapping("/{catId}")
    fun getCatByIdI(@PathVariable catId: Long): Mono<CatDto?>? {
        return catService.getCatById(catId)
    }

    @GetMapping("/full/{catId}")
    fun getFullCatById(@PathVariable catId: Long): Mono<CatDto?>? {
        return catService.getFullCatById(catId)
    }

    @get:GetMapping("/full")
    val fullAllCats: Flux<CatDto?>?
        get() = catService.fullAllCats

    @get:GetMapping("/fullForTest")
    val fullAllCatsReactiveForTest: Flux<CatDto?>
        get() = catService.fullAllCats.map { catDto: CatDto? ->
            try {
                Thread.sleep(1)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
            catDto
        }

    @get:GetMapping("/full/nonreactive")
    val fullAllCatsNonReactive: List<CatDto?>?
        get() {
            try {
                Thread.sleep(1)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
            return catService.fullAllCatsNonReactive
        }
}